### PR TITLE
fix: devcontainer and pytest updates to allow tests to run successfully

### PR DESCRIPTION
**Pull request by mohamed-cds**
# Summary
Fix running tests in VS Code devcontainer by:

* Enabling access to docker from within the container (primarily for redis tests);
* Bumping the postgres version to match the Github actions CI setup;
* Introduction of a `SQLALCHEMY_DATABASE_TEST_URI` to facilitate switching databases when running tests from inside the devcontainer; and
* Dynamically selecting the hostname for the redis server whether you're running locally or in the devcontainer.

### CODE REVIEW

File: .devcontainer/devcontainer.json [hunk 1]
```diff
@@ -13,9 +13,16 @@
 	},
 
 	"containerEnv": {
-		"SHELL": "/bin/bash"
+		"SHELL": "/bin/bash",
 	},
 
+  "features": {
```
Comment (patheard) [inline .devcontainer/devcontainer.json:19]: nit: formatting

**Review by patheard — CHANGES_REQUESTED** [review]
Good stuff - just had a formatting comment on my end.

File: .devcontainer/devcontainer.json (hunk 1 above)
Comment (mohamed-cds) [reply .devcontainer/devcontainer.json]: sneaky space snuck in there. Thanks!

File: .devcontainer/docker-compose.yml [hunk 2]
```diff
@@ -34,3 +36,9 @@ services:
       - "5432" 
     ports:
       - "5432:5432"
+  redis:
+    image: redis:6.2
+    restart: always
+    command: redis-server --port 6380
+    ports:
+      - "6380:6380"
```
Comment (jimleroyer) [inline .devcontainer/docker-compose.yml:44]: nice!

File: tests/conftest.py [hunk 3]
```diff
@@ -11,6 +11,13 @@
 from app import create_app, db
 
 
+def pytest_configure(config):
+    # Swap to test database if running from devcontainer
+    if os.environ.get("SQLALCHEMY_DATABASE_TEST_URI") is not None:
+        os.environ["SQLALCHEMY_DATABASE_URI"] = os.environ.get("SQLALCHEMY_DATABASE_TEST_URI")
+        os.environ["SQLALCHEMY_DATABASE_READER_URI"] = os.environ.get("SQLALCHEMY_DATABASE_TEST_URI")
```
Comment (jimleroyer) [inline tests/conftest.py:18]: Very nice!

**Review by jimleroyer — APPROVED** [review]

**Review by patheard — APPROVED** [review]

**Pull request merged**